plugins/search-react/src/context/SearchContext.test.tsx

```diff
@@ -421,4 +423,40 @@ describe('SearchContext', () => {
       });
     });
   });
+
+  describe('analytics', () => {
+    it('Captures analytics events if enabled in app', async () => {
```

Comment:
can you just add a test with the test case of not capturing analytics event if not enabled in app? to make sure the rest works as before?

Reply:
https://github.com/backstage/backstage/blob/0e8b015a35ada4720bd8d20c670ed7afb081cea1/packages/core-plugin-api/src/analytics/useAnalytics.tsx#L29

The analytics API is actually made to be safe to always call - whether anything is attached to it or not.

Reply:
maybe instead of changing the original tests (and wrapper) you can add a second wrapper for rendering with analytics api, and keep the wrapper as it is for the rest of the test cases

```
 const wrapperWithAnalytics = ({ children, initialState, config = {} }: any) => {
    const configApiMock = new MockConfigApi(config);
    return (
      <TestApiProvider
        apis={[
          [configApiRef, configApiMock],
          [searchApiRef, searchApiMock],
          [analyticsApiRef, analyticsApiMock],
        ]}
      >
        <SearchContextProvider initialState={initialState}>
          {children}
        </SearchContextProvider>
      </TestApiProvider>
    );
  };
```

Reply:
that's better, done!